File: 1909. Remove One Element to Make the Array Strictly Increasing/solution.py
class Solution(object):
  def canBeIncreasing(self, nums):
    """
    :type nums: List[int]
    :rtype: bool
    """
    found = False
    last = 0
    for index, num in enumerate(nums):
      if last < num:
        last = num
      else:
        if found:
          return False
        if index == 1 or nums[index - 2] < num:
          last = num
        found = True

    return True

File: 1909. Remove One Element to Make the Array Strictly Increasing/solution.cpp
class Solution {
public:
  bool canBeIncreasing(vector<int>& nums) {
    bool found = false;
    int last = 0;
    for (int index = 0; index < nums.size(); index++){
      int num = nums[index];
      if (last < num){
        last = num;
      }
      else{
        if (found){
          return false;
        }
        if (index == 1 || nums[index - 2] < num){
          last = num;
        }
        found = true;
      }
    }

    return true;
  }
};